core/modfunc.go

```diff
@@ -376,9 +371,10 @@ func (fn *ModuleFunction) Call(ctx context.Context, opts *CallOpts) (t dagql.Typ
 		}
 	}
 
-	if err := fn.linkDependencyBlobs(ctx, result, returnValueTyped); err != nil {
-		return nil, fmt.Errorf("failed to link dependency blobs: %w", err)
-	}
+	// NOTE: once generalized function caching is enabled we need to ensure that any non-reproducible
```

Comment:
Sounds good - note, that also this wasn't working in some cases anyways, because of the weird way that private inputs work: https://github.com/dagger/dagger/pull/8358#issue-2510229992